p2p/peer.go

```diff
@@ -247,12 +258,37 @@ func (p *peer) Status() tmconn.ConnectionStatus {
 	return p.mconn.Status()
 }
 
+// NewSend sends the message in the envelope on the channel specified by the
+// envelope. Returns false if the connection times out trying to place the message
+// onto its internal queue.
```

Comment:
It would be helpful here to have some kind of documentation that describes the additional metrics to which one's given access by using this method, as opposed to using `Send` (i.e. we should sell the benefits of using this method).

Reply:
Done.